### PR TITLE
feat(markdown_inline): Conceals for character escapes

### PR DESCRIPTION
**Pull request by ValdezFOmar**

### CODE REVIEW

**Comment by ribru17** [comment]
I think you could change all of the paragraphs to
```query
((backslash_escape) @string.escape
  (#offset! @string.escape 0 0 0 -1)
  (#set! conceal ""))
```
This also fixes the fact that the current queries are missing some other possible escape sequences, like `\$`

File: queries/markdown_inline/highlights.scm [hunk 1]
```diff
@@ -87,6 +87,11 @@
   (uri_autolink)
 ] @markup.link.url @nospell
 
+[
+  (link_label)
+  (email_autolink)
```
Comment (ribru17) [inline queries/markdown_inline/highlights.scm:92]: Looks like we also have no regular highlights for email autolinks. Would you mind giving it one?

File: queries/markdown_inline/highlights.scm [hunk 2]
```diff
@@ -87,6 +87,11 @@
...
+] @nospell
```
Comment (ribru17) [inline queries/markdown_inline/highlights.scm:93]: While we're at it, can you add `(entity_reference)` to the nospell list?

**Comment by clason** [comment]
To be honest, I'm not sure we want this (but also not sure yet we don't). The issue is that the markdown parser is used by Neovim as a renderer for LSP documentation, and it seems to me that such strings are something you would want to see literally (for -- manual -- copy&pasting). Unfortunately, conceal is all-or-nothing and already used for rendering, so we can't play the "just don't enable conceal" card here.

**Comment by clason** [comment]
Also, can you pull out the unrelated changes to a separate PR? The `@nospell` for `link_label` is also debatable, as the label often _is_ normal text that should be spell-checked.

**Comment by ValdezFOmar** [comment]
> Also, can you pull out the unrelated changes to a separate PR? The `@nospell` for `link_label` is also debatable, as the label often _is_ normal text that should be spell-checked.

Done https://github.com/nvim-treesitter/nvim-treesitter/pull/6750

**Comment by clason** [comment]
Closing since I still think we need to be careful about default conceal; I'd prefer to leave this driven by upstream. Thanks, though!

**Pull request closed without merging**